api/src/main/java/org/eclipse/microprofile/rest/client/RestClientBuilder.java

```diff
@@ -50,11 +51,36 @@ static RestClientBuilder newBuilder() {
      * <code>http://my-service:8080/service/api</code> in addition to any
      * <code>@Path</code> annotations included on the method.
      *
+     * The {@link #baseUri(URI)} method should be preferred over this method.
+     *
+     * Subsequent calls to this method will replace the previously specified
+     * baseUri/baseUrl.
+     *
      * @param url the base Url for the service.
      * @return the current builder with the baseUrl set
+     * @deprecated use {@link #baseUri(java.net.URI)} instead.
      */
+    @Deprecated
```

Comment:
I don't know that I agree with deprecating it.  I agree with adding support for URI objects, but it seems a bit extreme to say that this is now deprecated.  FWIW, I have 0 recollection on how we ended up on URL; I think I had assumed that `URL` was supported across the board.

Reply:
Right, I don't remember discussing it either.  I saw that other proxy-based REST client implementations also use URLs, so maybe that is why.  The argument to deprecate is driven by the network activity issues mentioned above.  There is also the issue that constructing URLs requires that the user catch a MalformedURLException.  Users could use `URI.create(someUrl)` to create a URI without needing to catch any exceptions.